### PR TITLE
Init dev

### PR DESCRIPTION
**Pull request by PTKu**

### CODE REVIEW

File: src/TcoCore/src/TcoCore.Wpf/TcoTask/TcoTaskViewModel.cs [hunk 1]
```diff
@@ -11,8 +11,9 @@ public class TcoTaskViewModel : RenderableViewModel
     {
         public TcoTaskViewModel()
         {
-            ResetCommand = new RelayCommand(a => this.TcoTask._taskState.Cyclic = (int)eTaskState.Idle);
-            InvokeCommand = new RelayCommand(a => this.TcoTask._taskState.Cyclic = (int)eTaskState.Request);
+
```
Comment (PTKu) [inline src/TcoCore/src/TcoCore.Wpf/TcoTask/TcoTaskViewModel.cs:14]: @TomKovac _taskState is 'hidden'? we should make it [ReadOnly()].

**Comment by PTKu** [comment]
#12 
# Messenger
## Architecture/structure
- ```TcoMessenger``` is for static messaging (does not implement fast logging yet)
- Each ```TcoObject``` implements TcoMessenger
- ```TcoMessenger``` contains ```mime``` Most Important Message

## Operation

* ```MiMe``` contains the message with highest priority for given object. 
* When more than one message is raised on the same object in the same cycle, the message with higher priority is stored as ```MiMe```
* When there are more messages in single cycle with the same priority the first incoming message is stored as ```MiMe```

## Usage

~~~
FUNCTION_BLOCK fbPistonMoveTask EXTENDS TcoCore.TcoTask
VAR_INPUT
	PositionSensor : BOOL;
END_VAR
VAR_IN_OUT
	MoveToPositionSignal : BOOL;
	InverseSignal : BOOL;	
END_VAR
//---------------------------------------------------------------------------
IF(THIS^.Execute())THEN
	THIS^.Messenger.Debug('Executing piston movement');
	InverseSignal := FALSE;
	MoveToPositionSignal := TRUE;
	THIS^.DoneWhen(PositionSensor);
END_IF
~~~

**Review by TomKovac — APPROVED** [review]

**Comment by PTKu** [comment]
https://github.com/Inxton/TcOpen/pull/18/commits/4e94faf496f285fc0aec8a8426f81d1ec3a8a403

#12 

- Messenger performance tests test, we get about 100 active messages in 0.5 sec to the UI at this point (room for improvement)

**Comment by PTKu** [comment]
Moving to dev... pipeline testing

**Pull request merged**